7_react/react1/week3/ToDoList/src/Components/TaskInput.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import "../index.css";
+import FetchTask from "./FetchTask";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+
+function TaskInput() {
+  const [toDoList, setToDoList] = useState([]);
+  const [todo, setTodo] = useState("");
+  const [selectedDate, setSelectedDate] = useState(new Date());
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    const newTodo = {
+      id: new Date().getTime(),
+      text: todo,
+      completed: false,
+      deadline: selectedDate,
+    };
+    setToDoList((prevTodos) => [...prevTodos, newTodo]);
+    setTodo("");
+  }
+
+  function deleteTodo(id) {
+    const updateToDoList = toDoList.filter((todo) => todo.id !== id);
+    setToDoList(updateToDoList);
+  }
+  function checkTodo(id) {
+    const checkToDoList = toDoList.map((todo) => {
+      if (todo.id === id) {
+        todo.completed = !todo.completed;
+      }
+      return todo;
+    });
+    setToDoList(checkToDoList);
+  }
+
+  return (
+    <div className="App">
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          placeholder="Add Item.."
+          onChange={(e) => setTodo(e.target.value)}
+          value={todo}
+        />
+        <DatePicker
+          selected={selectedDate}
+          onChange={(date) => setSelectedDate(date)}
+          dateFormat="dd/MM/yyyy"
+        />
+        <button className="add-button" type="submit">
+          Add
+        </button>
+      </form>
+      <FetchTask />
```

Comment:
I would suggest changing the name of the component, something like `TaskList` is more common

Reply:
Also, why is the list inside of the TaskInput? Either it should be renamed e.g TaskForm, or moved out